src/proof_of_possession.rs

```diff
@@ -346,6 +335,52 @@ mod test {
         .unwrap();
     }
 
+    #[tokio::test]
+    async fn basic_didkey_p256() {
+        let expires_in = Duration::minutes(5);
+        let jwk = JWK::generate_p256();
+        let did = DIDKey::generate(&jwk).unwrap();
```

Comment:
You can use `DIDKey::generate_url` instead of resolving and extracting the URL yourself.